Applications/IK/test/testIK.cpp

```diff
@@ -67,3 +73,99 @@ int main()
     cout << "Done" << endl;
     return 0;
 }
+
+void testMarkerWeightAssignments(const std::string& ikSetupFile)
+{
+    // Check the consistency for the IKTaskSet in the IKTool setup
+    InverseKinematicsTool ik(ikSetupFile);
+
+    // Get a copy of the IK tasks
+    IKTaskSet tasks = ik.getIKTaskSet();
+
+    // assign different weightings so we can verify the assignments
+    int nt = tasks.getSize();
+    for (int i{ 0 }; i < nt; ++i) {
+        tasks[i].setWeight(double(i));
+    }
+
+    cout << tasks.dump(true) << endl;
+    // update tasks used by the IK tool
+    ik.getIKTaskSet() = tasks;
+
+    // perform the check
+    checkMarkersReferenceConsistencyFromTool(ik);
+
+    // Now reverse the order and half the number of tasks
```

Comment:
Actually, the code below doesn't result in half as many tasks when `nt` is odd (it will result in `(nt+1)/2`). Also see comment on L104. Perhaps simply "...and reduce the number of tasks..." since it isn't important exactly how many tasks remain.